service/history/api/reapplyevents/api.go

```diff
@@ -195,13 +187,10 @@ func Invoke(
 				shard.GetLogger().Error("failed to re-apply stale events", tag.Error(err))
 				return nil, err
 			}
-			if len(reappliedEvents) == 0 {
-				return &api.UpdateWorkflowAction{
-					Noop:               true,
-					CreateWorkflowTask: false,
-				}, nil
-			}
-			return postActions, nil
+			return &api.UpdateWorkflowAction{
+				Noop:               true,
+				CreateWorkflowTask: false,
+			}, nil
```

Comment:
this section of logic will skip the persistence of mutable state if events are reapplied?